src/main/java/frc/trigon/robot/subsystems/ledstrip/LEDStrip.java

```diff
@@ -0,0 +1,146 @@
+                slotID
+        );
+    }
+
+    void animateColourFlow(Color color, double speed, ColorFlowAnimation.Direction direction) {
+        CANDLE.animate(new ColorFlowAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        speed,
+                        this.numLEDs,
+                        direction,
+                        this.offset
+                ),
+                slotID
+        );
+    }
+
+    void animateLarson(Color color, double speed, LarsonAnimation.BounceMode mode, int size) {
+        CANDLE.animate(
+                new LarsonAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        speed,
+                        this.numLEDs,
+                        mode,
+                        size,
+                        this.offset),
+                slotID
+        );
+    }
+
+    void animateRGBFade(double brightness, double speed) {
+        CANDLE.animate(
+                new RgbFadeAnimation(
+                        brightness,
+                        speed,
+                        this.numLEDs,
+                        this.offset),
+                slotID
+        );
+    }
+
+    void animateSingleFade(Color color, double speed) {
+        CANDLE.animate(
+                new SingleFadeAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        speed,
+                        this.numLEDs,
+                        this.offset),
+                slotID
+        );
+    }
+
+    void animateTwinkle(Color color, double speed, TwinkleAnimation.TwinklePercent divider) {
+        CANDLE.animate(
+                new TwinkleAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        speed,
+                        this.numLEDs,
+                        divider,
+                        this.offset
+                ),
+                slotID
+        );
+    }
+
+    void animateStrobe(Color color, double speed) {
+        CANDLE.animate(
+                new StrobeAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        speed,
+                        this.numLEDs,
+                        this.offset
+                ),
+                slotID
+        );
+    }
+
+    void clearAnimation() {
+        CANDLE.clearAnimation(slotID);
+    }
+
+    private int setId() {
+        lastCreatedLEDStrip++;
+        return lastCreatedLEDStrip;
+    }
```

Comment:
A function should do one thing. Either return a value or interact with one. Choose what you want to do here.